src/bloqade/squin/analysis/rewrite/stim.py

```diff
@@ -0,0 +1,30 @@
+from typing import Dict
+from dataclasses import dataclass
+
+from bloqade.analysis.address import Address
+from bloqade.squin.analysis.nsites import Sites
+
+from kirin import ir
+from kirin.rewrite.abc import RewriteResult, RewriteRule
+
+
+@dataclass
+class SquinToStim(RewriteRule):
+    
+    # Somehow need to plug in Address and Sites
+    # into the SSAValue Hints field, which only accepts
+    # Attribute types
+
+    ## Could literally just plug in `ir.Attribute` into
+    ## the Address and Site lattices? 
+    ## Couldn't I just create my own attributes instead?
+
+    address_analysis: Dict[ir.SSAValue, Address]
+    op_site_analysis: Dict[ir.SSAValue, Sites] 
```

Comment:
I think you would want to store these analysis results in the SSAValue hints. They are useful for other passes too.

Reply:
Yes! Just taking inspiration from the `WrapConst` rewrite, I'm just thinking over how to get `Address` and `Sites` to be `Attributes` because of the type `hints` can accept.